### PR TITLE
pythonPackages.mal: init at 2018-03-01

### PR DESCRIPTION
**Pull request by taktoa**
###### Motivation for this change

Adds the [mal](https://github.com/ryukinix/mal) Python package.

###### Things done

- [ ] Tested using sandboxing
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/decorating/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, buildPythonPackage, fetchPypi, isPy27, isPy34, isPy35, isPy36 }:
+
+assert (isPy27 || isPy34 || isPy35 || isPy36);
```
Comment (FRidh) [inline pkgs/development/python-modules/decorating/default.nix]: this should be removed and replaced with a `disabled = ...;` argument passed to `buildPython*`

File: pkgs/development/python-modules/decorating/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+buildPythonPackage rec {
+  name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/decorating/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/development/python-modules/mal/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchFromGitHub
+, buildPythonPackage, isPy34, isPy35
+, appdirs, requests, decorating
+}:
+
+# tests currently fail on Python 3.6, try adding isPy36 when the next version
+# after 0.3.2 is released
+assert (isPy34 || isPy35);
```
Comment (FRidh) [inline pkgs/development/python-modules/mal/default.nix]: same here

File: pkgs/development/python-modules/mal/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+buildPythonPackage rec {
+  name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/mal/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

**Review by FRidh — CHANGES_REQUESTED** [review]

**Review by dotlambda — CHANGES_REQUESTED** [review]
`mal` is not a library and should therefore use `buildPythonApplication` and be moved out of `pythonPackages`.

**Comment by JohnAZoidberg** [comment]
The package seems to not work anymore so we don't need it.
See: https://github.com/ryukinix/mal

**Comment by aanderse** [comment]
@taktoa can you please confirm if this package is no longer functional and close if so?

**Pull request closed without merging**